lib/helpers/query/castUpdate.js

```diff
@@ -42,6 +41,11 @@ module.exports = function castUpdate(schema, obj, options, context, filter) {
     return obj;
   }
 
+  if (schema.options.strict === 'throw') {
+    throw new StrictModeError();
+  } else if (!('overwriteDiscriminatorKey' in context._mongooseOptions)) {
```

Comment:
Should use [`hasOwnProperty()`](https://masteringjs.io/tutorials/fundamentals/hasownproperty) instead.

Reply:
Also need to check if `context._mongooseOptions != null`